packages/popover/src/index.js

```diff
@@ -0,0 +1,246 @@
+                    innerRef={this.contentRef}
+                    {...styleProps}
+                    tabIndex="-1"
+                  >
+                    <Box id={`popover-description-${this.props.idx}`}>
+                      {this.props.renderContent({
+                        handleClose: this.handleClose
+                      })}
+                    </Box>
+                  </ContentContainer>
+                  <Arrow
+                    innerRef={arrowProps.ref}
+                    style={arrowProps.style}
+                    data-placement={placement}
+                    {...styleProps}
+                    aria-hidden="true"
+                  />
+                </PopperGuide>
+              )}
+            </Popper>,
+            // Append each instance of the Popover as portal directly to the body
+            document.querySelector('body')
+          )}
+      </Manager>
+    )
+  }
+}
+
+const PopperGuide = styled.div`
+  padding: 16px;
+`
+const ContentContainer = styled.section`
+  box-shadow: ${({ theme }) => theme.boxShadows[1]};
+  font-size: ${({ theme }) => theme.fontSizes[0]}px;
+  border-radius: ${({ theme }) => theme.radii[1]}px;
+  background: ${({ theme, background }) => theme.colors[background]};
+  box-sizing: border-box;
+  outline: 0;
+  max-width: 100%;
+`
+
+const ArrowAlignment = () =>
+  `
+    &[data-placement*="right"] {
+      left: 0;
+    }
+    &[data-placement*="left"] {
+      right: 0;
+    }
+    &[data-placement*="top"] {
+      bottom: 0;
+    }
+    &[data-placement*="bottom"] {
+      top: 0;
+    }
+    &[data-placement*="bottom"]::before,
+    &[data-placement*="right"]::before,
+    &[data-placement*="top"]::after,
+    &[data-placement*="left"]::after {
+      content: '';
+      position: absolute;
+      width: 0;
+      height: 0;
+      margin: auto;
+      border-style: solid;
+      transform-origin: 0 0;
+      transform: rotate(-45deg);
+      border-width: 5px;
+    }
+  `
+
+const ArrowPlacement = ({ theme, background }) => {
+  const color = theme.colors[background]
+  const shadow = `rgba(0,0,0,0.08)`
+
+  return `
+    &[data-placement*="right"]::before {
+      left: 9px;
+      border-color: ${color} transparent transparent ${color};
+      box-shadow: -2px -2px 2px 0px ${shadow};
+    }
+    &[data-placement*="left"]::after {
+      right: 14px;
+      border-color: transparent ${color} ${color} transparent;
+      box-shadow: 2px 2px 2px 0px ${shadow};
+    }
+    &[data-placement*="top"]::after {
+      top: -16px;
+      margin-left: -5px;
+      border-color: transparent transparent ${color} ${color};
+      box-shadow: -2px 2px 2px 0px ${shadow};
+    }
+    &[data-placement*="bottom"]::before {
+      top: 16px;
+      margin-left: -5px;
+      border-color: ${color} ${color} transparent transparent;
+      box-shadow: 2px -2px 2px 0px ${shadow};
+    }
+  `
+}
+const Arrow = styled.span`
+  position: absolute;
+  ${ArrowAlignment}
+  ${ArrowPlacement}
+`
```

Comment:
Could you use different file(s) for the Arrow* component(s) for modularity, i.e., another component could import to share Arrow* components as well.